core/schema.py

```diff
@@ -263,9 +263,19 @@ def mutate_and_get_payload(cls, root, info, **data):
                 try:
                     mutation_data = data.copy()
                     mutation_data.pop("mutation_extensions", None)
-                    error_messages = cls.async_mutate(
-                        info.context.user if info.context and info.context.user else None,
-                        **mutation_data)
+                    if mutation_data.get('autogenerate_code', None):
+                        returned = cls.async_mutate(info.context.user if info.context and info.context.user else None,
+                                                    **mutation_data)
+
+                        if isinstance(returned, Dict):
+                            data.update(returned)
+                            error_messages = None
+                        else:
+                            error_messages = returned
+                    else:
+                        error_messages = cls.async_mutate(
+                            info.context.user if info.context and info.context.user else None,
+                            **mutation_data)
```

Comment:
This code can be expressed as:
```suggestion
                    messages = cls.async_mutate(info.context.user if info.context and info.context.user else None, **mutation_data)
                    if mutation_data.get('autogenerate_code', False) and isinstance(messages, Dict):
                        data.update(messages)
                        error_messages = None
                    else:
                        error_messages = messages
```